src/ML/src/MANNAutoregressive.cpp

```diff
@@ -22,10 +22,70 @@
 using namespace BipedalLocomotion::ML;
 using namespace BipedalLocomotion;
 
+MANNFootState MANNFootState::generateFootState(iDynTree::KinDynComputations& kinDyn,
+                                               const std::vector<Eigen::Vector3d>& corners,
+                                               const std::string& footName,
+                                               int footIndex)
+{
+    using namespace BipedalLocomotion::Conversions;
+
+    MANNFootState dummyState;
+    // the schmitt trigger is initialized to true since we assume that the foot is in contact
+    dummyState.schmittTriggerState.state = true;
+    dummyState.contact.name = footName;
+    dummyState.contact.index = footIndex;
+    dummyState.contact.isActive = dummyState.schmittTriggerState.state;
+    dummyState.contact.switchTime = std::chrono::nanoseconds::zero();
+    dummyState.contact.pose = toManifPose(kinDyn.getWorldTransform(footIndex));
+
+    // we need to force the height of the foot to be zero.
+    dummyState.contact.pose.translation(Eigen::Vector3d{dummyState.contact.pose.translation()[0],
+                                                        dummyState.contact.pose.translation()[1],
+                                                        0.0});
+    dummyState.corners = corners;
+    return dummyState;
+}
+
+MANNAutoregressive::AutoregressiveState
+MANNAutoregressive::AutoregressiveState::generateDummyAutoregressiveState(
+    const MANNInput& input,
+    const MANNOutput& output,
+    const manif::SE3d& I_H_B,
+    const MANNFootState& leftFootState,
+    const MANNFootState& rightFootState)
+{
+    MANNAutoregressive::AutoregressiveState autoregressiveState;
+    // 51 is the length of 1 second of past trajectory stored in the autoregressive state. Since the
+    // original mocap data are collected at 50 Hz, and therefore the trajectory generation is
+    // assumed to proceed at 50 Hz, we need 50 datapoints to store the past second of trajectory.
+    // Along with the present datapoint, they sum up to 51!
+    constexpr size_t lengthOfPresentPlusPastTrajectory = 51;
```

Comment:
Nice explanation. Something to understand would be the likelihood that in a recent future, we end up acquiring mocap data at a different frequency.

Reply:
@paolo-viceconte  I Choose You! :)

Reply:
We can actually allow to set this as a configuration parameter! 

Let's keep in mind that here different parameters are related as follows:
- the network input is defined to include [`projected_base_horizon`](https://github.com/ami-iit/bipedal-locomotion-framework/blob/ae629760046a4d5cc01c897d12846254aaa80b00/examples/MANN/config/config_mann.toml#L53C1-L53C23) (in our case, 12) base positions projected on the ground, of which `projected_base_horizon / 2` (in our case, 6) encode the past base motion and `projected_base_horizon / 2` (in our case, 6) the future one. Here there is a hidden constraint for `projected_base_horizon` to be even.
- the aforementioned `projected_base_horizon / 2` past base positions projected on the ground are subsampled from `lengthOfPresentPlusPastTrajectory` datapoints which store the last 1 second of base motion at the data frequency (in our case, the last 50 + 1 datapoints)

These two configuration parameters are decoupled, meaning that one could decide to train a network which takes 20 equally-spaced datapoints on a 1s time window or 30 equally-spaced datapoints on a 10s time window, etc. So we can allow as new configuration parameters:
- `past_projected_base_datapoints` or `future_projected_base_datapoints` (in our case, 6)  
- `mocap_data_frequency` (in our case, 50) 
- `past_projected_base_horizon` (in our case, 1) as number of seconds to look back in the past (as an integer? or float but then we need to round somewhere)

So that then:
- the old `projected_base_horizon` becomes `projected_base_datapoints = 2 * past_projected_base_datapoints` (so to enforce it to be even)
- `lengthOfPresentPlusPastTrajectory = past_projected_base_horizon * mocap_data_frequency + 1`

Reply:
> past_projected_base_datapoints or future_projected_base_datapoints (in our case, 6)
mocap_data_frequency (in our case, 50)
past_projected_base_horizon (in our case, 1) as number of seconds to look back in the past (as an integer? or float but then we need to round somewhere)


I will use this names